dbms/src/Columns/Collator.cpp

```diff
@@ -26,16 +30,75 @@ namespace DB
     }
 }
 
+
+std::unique_ptr<AvailableCollationLocales> AvailableCollationLocales::instance_impl;
+std::once_flag AvailableCollationLocales::init_flag;
+
+void AvailableCollationLocales::init()
+{
+    instance_impl = std::make_unique<AvailableCollationLocales>();
+#if USE_ICU
+    size_t available_locales_count = ucol_countAvailable();
+    for (size_t i = 0; i < available_locales_count; ++i)
+    {
+        std::string locale_name = ucol_getAvailable(i);
+        UChar lang_buffer[128]; /// 128 is enough for language name
+        char normal_buf[128];
+        UErrorCode status = U_ZERO_ERROR;
+        /// All names will be in English language
+        size_t lang_length = uloc_getDisplayLanguage(locale_name.c_str(), "en", lang_buffer, 128, &status);
+        if (U_FAILURE(status))
+            instance_impl->available_collation_locales.push_back(LocaleAndLanguage{locale_name, "unknown"});
+        else
+        {
+            u_UCharsToChars(lang_buffer, normal_buf, lang_length);
```

Comment:
Please add comment about what this function does.
I've found it in ICU docs, but it is not obvious enough. Also not obvious that the result string has the same length (but true).